backend/LeetWars.Builder/Services/SolutionRunnerService.cs

```diff
@@ -1,29 +1,105 @@
 ﻿using Docker.DotNet;
 using Docker.DotNet.Models;
+using LeetWars.Builder.Helpers.BuildResultReader;
+using LeetWars.Builder.Helpers.DirectoryConfigurations;
+using LeetWars.Builder.Helpers.DockerConfigurations;
+using LeetWars.Builder.Helpers.RunnerFileWriterReader;
 using LeetWars.Builder.Interfaces;
+using LeetWars.Builder.Models;
+using LeetWars.Builder.Models.HelperModels;
+using Newtonsoft.Json;
 
 namespace LeetWars.Builder.Services
 {
     public class SolutionRunnerService : ISolutionRunnerService, IDisposable
     {
         private readonly DockerClient _client = new DockerClientConfiguration().CreateClient();
 
-        public async Task RunCodeInContainerAsync(string imageName, string containerName)
+        public async Task<CodeRunResults> Run(CodeRunRequest request)
         {
-            var containerParams = new CreateContainerParameters
+            var buildResultReader = new BuildResultReader();
+            var dirBuilder = new DockerMountDirectoryConfigurations();
+            var dockerConfig = new DockerConfigurations();
+            RunnerFileWriterReaderClass readWrite = new RunnerFileWriterReaderClass();
+            dirBuilder.RFW = readWrite;
+
+            string dir = Directory.GetCurrentDirectory();
+            Config config;
+
+            string? solutionFileName;
+            switch (request.Language)
             {
-                Image = imageName,
-                Name = containerName
-            };
+                case Languages.csharp:
+                    var projectFileName = await dirBuilder.CreateDirectoryCSharp(dir, request);
+
+                    await dockerConfig.CreateImage(_client, SDKImageNames.CSharpSDK);
+
+                    solutionFileName = SolutionFileNaming.CSharpSolutionName;
+                    config = dockerConfig.GetCSharpConfig(projectFileName);
+                    break;
+
+                case Languages.javascript:
+                    dirBuilder.CreateDirectoryJS(dir, request);
+
+                    await dockerConfig.CreateImage(_client, SDKImageNames.JSSDK);
 
-            var container = await _client.Containers.CreateContainerAsync(containerParams);
+                    solutionFileName = SolutionFileNaming.JSSolutionName;
+                    config = dockerConfig.GetJSConfig(solutionFileName);
+                    break;
+                default:
+                    return new CodeRunResults();
+            }
```

Comment:
to a separate method

Reply:
and 2 cases are almost the same, so you can combine them into one and just pass specific parameters